robot_framework/process.py

```diff
@@ -1,8 +1,343 @@
+    """Get all emails to be handled by the robot.
+
+    Args:
+        graph_access: The GraphAccess object used to authenticate against Graph.
+
+    Returns:
+        A filtered list of email objects to be handled.
+    """
+    # Get all emails from the relevant folder.
+    mails = graph_mail.get_emails_from_folder("itk-rpa@mkb.aarhus.dk", config.MAIL_SOURCE_FOLDER, graph_access)
+
+    # Filter the emails on sender and subject
+    mails = [mail for mail in mails if mail.sender == "noreply@aarhus.dk" and mail.subject == config.MAIL_INBOX_SUBJECT]
+
+    return mails
+
+
+def _parse_mail_text(mail_text: str) -> dict:
+    ''' From an email sent by the OS2 Forms form "Masseoprettelse i KMD Nova",
+    create a new dictionary.
+
+    Args:
+        mail_text: Text from OS2 Forms that can be read by the robot.
+
+    "Returns:
+        A dictionary object, for easier access to variables.
+    '''
+    dictionary = html_to_dict(mail_text)
+    if dictionary["Brug eksisterende sag"] != "Valgt":
+        dictionary["Følsomhed"] = _get_sensitivity_from_email(dictionary["Følsomhed"])
+        dictionary["Sikkerhedsenhed"] = _get_securityunit_from_department(dictionary["Afdeling"])
+    return dictionary
+
+
+def _get_sensitivity_from_email(email_string: str) -> str:
+    '''Translates the string from OS2 forms to the expected KMD Nova format.
+
+    Args:
+        email_string: A string, matching the format in OS2 Forms and the emails sent to the robot.
+
+    Returns:
+        A string, matching the format expected by KMD Nova in the Sensitivity field.
+    '''
+    return config.KMD_SENSITIVITY[email_string]
+
+
+def _get_securityunit_from_department(email_string: str) -> str:
+    '''Translates the string from OS2 Department value to a security unit value
+
+    Args:
+        email_string: A string, matching the format in OS2 Forms and the emails sent to the robot.
+
+    Returns:
+        A string, matching the format expected by KMD Nova in the Security field. All options except two will return Borgerservice.
+    '''
+    return config.KMD_DEPARTMENT_SECURITY_PAIR[email_string]
+
+
+def _get_name_from_cpr(cpr: str, nova_access: NovaAccess) -> str:
+    '''Find name from a matching case or lookup by address.
+
+    Args:
+        cases: A list of NovaCases to check for existing name.
+        cpr: ÍD of the person we are looking for.
+        nova_access: A token to access the KMD Nova API.
+
+    Returns:
+        The name of the person with the provided CPR.
+    '''
+    cases = nova_cases.get_cases(nova_access, cpr = cpr, limit=1)
+    # Check provided cases for an existing match with a name:
+    for case in cases:
+        for case_party in case.case_parties:
+            if case_party.identification == cpr and case_party.name:
+                return case_party.name
+    # If nothing was found, do a lookup in the registry:
+    return nova_cpr.get_address_by_cpr(cpr, nova_access)['name']
```

Comment:
Why not just do this?

Reply:
Because our test CPRs doesn't exist in the registry, so we need to get their names from Nova? It seems silly, I've moved the address check up, to avoid doing more lookups in Nova than we have to. I haven't tested if we really do need names to create new cases, but I suspect we do.